### DISCUSSION TITLE
Ordering and filtering in joined queries

### DISCUSSION THREAD

**Discussion by alexmakarchikov** [groue/GRDB.swift#1156] [Q&A] [answered]
Hi! I need to join queries and sorting(filtering) the result. I use the [codable way](https://github.com/groue/GRDB.swift#splitting-rows-the-codable-way). How can I sort by "player.name" column for example? The code below dont work
`static func fetchAllAndSortByName(_ db: Database) throws -> [PlayerInfo] {
        try all().sort(Column("player.name")).fetchAll(db)
    }`

**Comment by groue** [comment, marked as answer]
Hello @alexmakarchikov,

The "Splitting Rows" chapters are about manually-crafted SQL queries.

It looks like your question is about the query interface requests, that you define in Swift, which generate SQL queries for you.

So check the documentation about the `order` method:

- In [requests](https://github.com/groue/GRDB.swift/blob/master/README.md#requests)
- In [associations](https://github.com/groue/GRDB.swift/blob/master/Documentation/AssociationsBasics.md#sorting-associations)

After you have read the documentation, try to apply it. And if you still can't do what you need to do, ask a detailed question (please take inspiration from the [issue template](https://raw.githubusercontent.com/groue/GRDB.swift/master/.github/ISSUE_TEMPLATE.md)).

**Reply by alexmakarchikov** [reply]
Thanks! I  figured out how to do it
